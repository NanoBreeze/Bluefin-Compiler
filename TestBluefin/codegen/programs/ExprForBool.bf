/*
Verifies basic expressions for bools, including &&, ||, !, and true/false
*/

void hi(bool a, bool b) {
	a && b;
	a || b;

	a && true && b; // we expect the a && true to be folded into 'a'
	b || false || a;  // we expect the b || false to be folded into 'b'

	!a;
}