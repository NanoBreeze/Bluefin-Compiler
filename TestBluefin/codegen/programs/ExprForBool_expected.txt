; ModuleID = 'ExprForBool.bf'
source_filename = "ExprForBool.bf"

define void @hi(i1 %a, i1 %b) {
entry:
  %alloctmp_a = alloca i1, align 1
  store i1 %a, i1* %alloctmp_a, align 1
  %alloctmp_b = alloca i1, align 1
  store i1 %b, i1* %alloctmp_b, align 1
  %loadtmp_a = load i1, i1* %alloctmp_a, align 1
  %loadtmp_b = load i1, i1* %alloctmp_b, align 1
  %andtmp = and i1 %loadtmp_a, %loadtmp_b
  %loadtmp_a1 = load i1, i1* %alloctmp_a, align 1
  %loadtmp_b2 = load i1, i1* %alloctmp_b, align 1
  %ortmp = or i1 %loadtmp_a1, %loadtmp_b2
  %loadtmp_a3 = load i1, i1* %alloctmp_a, align 1
  %loadtmp_b4 = load i1, i1* %alloctmp_b, align 1
  %andtmp5 = and i1 %loadtmp_a3, %loadtmp_b4
  %loadtmp_b6 = load i1, i1* %alloctmp_b, align 1
  %loadtmp_a7 = load i1, i1* %alloctmp_a, align 1
  %ortmp8 = or i1 %loadtmp_b6, %loadtmp_a7
  %loadtmp_a9 = load i1, i1* %alloctmp_a, align 1
  %nottmp = xor i1 %loadtmp_a9, true
  ret void
}
